src/app/ReadClient.cpp

```diff
@@ -894,32 +917,58 @@ void ReadClient::OnLivenessTimeoutCallback(System::Layer * apSystemLayer, void *
                  "Subscription Liveness timeout with SubscriptionID = 0x%08" PRIx32 ", Peer = %02x:" ChipLogFormatX64,
                  _this->mSubscriptionId, _this->GetFabricIndex(), ChipLogValueX64(_this->GetPeerNodeId()));
 
+    if (_this->mIsPeerLIT)
+    {
+
+        // If a LIT ICD device enters an inactive state, the associated subscription is marked as `InactiveICDSubscription`.
+        // The ReadClient consumer then determines whether to initiate resubscription through the `OnResubscriptionNeeded` callback.
+        // This decision process is triggered by either a liveness timeout or the arrival of a check-in message.
+        // In the case of a liveness timeout, CHIP_ERROR_LIT_SUBSCRIBE_INACTIVE_TIMEOUT is passed to application and indicate LIT
+        // ICD device is inactive. In the case of a check-in message, the `OnActiveModeNotification` callback is invoked and
+        // the `InactiveICDSubscription` state is specifically checked. Note that the liveness timeout duration always exceeds
+        // the maximum interval time (and hence the idle duration). Therefore, transitioning the device to `InactiveICDSubscription`
+        // upon reaching the liveness timeout is safe and consistent.
+        ChipLogProgress(DataManagement, "Peer is not active now, mark the subscription as InactiveICDSubscription.");
+        _this->MoveToState(ClientState::InactiveICDSubscription);
+
+        // Mark defunct immediately as usual flow does
```

Comment:
This comment does not obviously have anything to do with this code.

Maybe what this should all document is that we call TriggerResubscriptionForLivenessTimeout as we normally would, but with an error code that makes it possible to tell that this is a LIT and that hence resubscribing directly is not likely to work, so that the implementation of TriggerResubscriptionForLivenessTimeout can act accordingly.

All the other bits about InactiveICDSubscription, what "act accordingly" means etc don't seem to belong here.

Which raises the question: should the transition to the InactiveICDSubscription state actually be in TriggerResubscriptionForLivenessTimeout?  Because maybe our API client in fact wants us to resubscribe immediately because it knows the LIT is active or something...

Reply:
After moving the MoveToState, I think we can delete these comments.